### PR TITLE
[5.0] Update `core.paginator`

### PR DESCRIPTION
**Pull request by Viicos**
Part of #1493.

### CODE REVIEW

File: django-stubs/core/paginator.pyi [hunk 1]
```diff
@@ -19,6 +19,8 @@ class _SupportsPagination(Protocol[_T], Sized, Iterable):
 
 class Paginator(Generic[_T]):
     ELLIPSIS: ClassVar[_StrPromise]
+    default_error_messages: ClassVar[dict[str, _StrPromise]]
+    error_messages: dict[str, _StrOrPromise]
```
Comment (flaeppe) [inline django-stubs/core/paginator.pyi]: I think we should declare these the same as model fields does. We can use the alias `django.db.models.fields._ErrorMessagesDict`

```suggestion
    default_error_messages: _ErrorMessagesDict
    error_messages: _ErrorMessagesDict
```

File: django-stubs/core/paginator.pyi [hunk 2]
```diff
@@ -29,6 +31,7 @@ class Paginator(Generic[_T]):
         per_page: int | str,
         orphans: int = ...,
         allow_empty_first_page: bool = ...,
+        error_messages: dict[str, _StrOrPromise] | None = ...,
```
Comment (flaeppe) [inline django-stubs/core/paginator.pyi]: Same here, but with the alias `django.db.models.fields._ErrorMessagesMapping`

```suggestion
        error_messages: _ErrorMessagesMapping | None = ...,
```

**Review by flaeppe — COMMENTED** [review]
My suggested changes might justify movement of the aliases currently living in `django.db.models.fields`, since they would now be used under a different domain. But that can be done separately, if we'd deem it relevant.

**Comment by Viicos** [comment]
I see `_ErrorMessagesDict` is imported in other files as well, e.g. from `contrib/auth/forms.pyi`, so I think it's fine to leave the type alias where it is currently. Then we might decide to move it at some point

File: django-stubs/core/paginator.pyi [hunk 3]
```diff
@@ -19,6 +20,8 @@ class _SupportsPagination(Protocol[_T], Sized, Iterable):
 
 class Paginator(Generic[_T]):
     ELLIPSIS: ClassVar[_StrPromise]
+    default_error_messages: ClassVar[_ErrorMessagesDict]
```
Comment (Viicos) [inline django-stubs/core/paginator.pyi:23]: values are actually guaranteed to be `_StrPromise`, but I think it's fine to leave it this way (`dict[str, _StrOrPromise]`)

**Review by flaeppe — APPROVED** [review]

File: django-stubs/core/paginator.pyi (hunk 3 above)
Comment (flaeppe) [reply django-stubs/core/paginator.pyi:23]: I suspect you'd then run in to troubles if you want to declare your own `default_error_messages` not bothering about translation strings. e.g.

```python
class X(Paginator):
    default_error_messages = {"key": "not a translated string"}
```

Comment (Viicos) [reply django-stubs/core/paginator.pyi:23]: Oh yes right

**Pull request merged**